runtime/src/governance/proposals.rs

```diff
@@ -320,6 +323,20 @@ decl_module! {
                 print(e);
             }
         }
+
+        /// Cancel a proposal and return stake without slashing
+        fn veto(proposal_id: u32) -> Result {
+            ensure!(<Proposals<T>>::exists(proposal_id), MSG_PROPOSAL_NOT_FOUND);
```

Comment:
Should we ensure that it was called by sudo key?

Reply:
this is handled automatically by decl_module macro when the method doesn't have an origin as first argument.